lib/websocket_lwt.ml

```diff
@@ -37,11 +37,35 @@ module Connected_client = struct
     ic: Request.IO.ic;
     oc: Request.IO.oc;
     http_request: Cohttp.Request.t;
-    standard_frame_replies: bool;
+    mutable standard_frame_replies: bool;
+    read_frame: unit -> Frame.t Lwt.t;
+    read_frame_std: unit -> Frame.t Lwt.t;
+    send_frame: Frame.t -> unit Lwt.t;
   }
 
-  let create random_string http_request flow ic oc =
-    let buffer = Buffer.create 128 in
+  let create ?(buffer=Buffer.create 128) random_string http_request flow ic oc =
+    let read_frame = make_read_frame ?random_string ~masked:false (ic, oc) in
```

Comment:
Here it would be incorrect to pass `buffer` to `make_read_frame`, since then the buffer would be shared between the reader and writer. Maybe the `buffer` argument to `create` should be renamed to `write_buffer` to make the distinction clear.